pkg/slack/events/mention/mention.go

```diff
@@ -0,0 +1,137 @@
+package mention
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+
+	"github.com/openshift/ci-tools/pkg/slack/events"
+	"github.com/openshift/ci-tools/pkg/slack/modals"
+	"github.com/openshift/ci-tools/pkg/slack/modals/bug"
+	"github.com/openshift/ci-tools/pkg/slack/modals/consultation"
+	"github.com/openshift/ci-tools/pkg/slack/modals/enhancement"
+	"github.com/openshift/ci-tools/pkg/slack/modals/helpdesk"
+	"github.com/openshift/ci-tools/pkg/slack/modals/incident"
+	"github.com/openshift/ci-tools/pkg/slack/modals/triage"
+)
+
+type messagePoster interface {
+	PostMessage(channelID string, options ...slack.MsgOption) (string, string, error)
+}
+
+// Handler returns a handler that knows how to respond to
+// new messages that mention the robot by showing users
+// which interactive workflows they might be interested in,
+// based on the phrasing that they used to mention the bot.
+func Handler(client messagePoster) events.PartialHandler {
```

Comment:
I tried to test this method but two things made it impossible:

1. the `*slackevents.EventsAPIEvent` type is _incredibly_ dynamic and does not implement custom JSON marshalling, so there's no way to round-trip it with a fixture (it does not serialize)
2. the `slack.Client.PostMessage` function - the only thing we really want to test - takes all of it's important arguments in a functional options pattern, which uses anonymous functions operating over unexported types to achieve what we need, and validating that the right options were passed is basically impossible without incredibly ugly reflection and other nonsense